src/impls.rs

```diff
@@ -113,200 +90,122 @@ impl_metadata_for_tuple!(A, B, C, D, E, F, G, H);
 impl_metadata_for_tuple!(A, B, C, D, E, F, G, H, I);
 impl_metadata_for_tuple!(A, B, C, D, E, F, G, H, I, J);
 
-impl<T> HasTypeId for Vec<T>
-where
-	T: Metadata + 'static,
-{
-	fn type_id() -> TypeId {
-		TypeIdCustom::new("Vec", Namespace::prelude(), tuple_meta_type![T]).into()
-	}
-}
-
-impl<T> HasTypeDef for Vec<T>
+impl<T> TypeInfo for Vec<T>
 where
 	T: Metadata + 'static,
 {
-	fn type_def() -> TypeDef {
-		TypeDefStruct::new(vec![NamedField::new("elems", MetaType::new::<[T]>())]).into()
+	fn type_info() -> Type {
+		TypeComposite::new("Vec", Namespace::prelude())
+			.type_params(tuple_meta_type![T])
+			.fields(Fields::named().field_of::<[T]>("elems"))
+			.into()
```

Comment:
Since we decided that we should communicate SCALE encoding via this crate this could be simplified to make `Vec` appear as its encoding, which is pretty much just the same as a Rust slice of `T`.

Reply:
I've made this change and it does solve the issue with `polkadot-js`. 

However we are also losing some information about the actual original type (what kind of collection it was) which might be useful in some cases?

Reply:
I only meant this for `Vec` and maybe `String` for now. Other collections such as `BTreeMap` etc. should remain "complex". However, we might think about adding support for native `mappings`. But before we do so we definitely need to discuss that with JS side.

Reply:
I've reverted `BTreeMap` to being "complex". However I think it requires more thought because the representation suggests it's a struct with an `elems` field - which is not right: an unnamed field might be better? `String` on the other hand is an actual struct with a `vec` field.